Mage/src/main/java/mage/abilities/common/CardsLeaveGraveyardTriggeredAbility.java

```diff
@@ -24,7 +26,15 @@ public CardsLeaveGraveyardTriggeredAbility(Effect effect) {
     }
 
     public CardsLeaveGraveyardTriggeredAbility(Effect effect, FilterCard filter) {
-        super(Zone.BATTLEFIELD, effect, false);
+        this(effect, filter, false);
+    }
+
+    public CardsLeaveGraveyardTriggeredAbility(Effect effect, boolean optional) {
+        this(effect, StaticFilters.FILTER_CARD_CARDS, false);
```

Comment:
you have to actually use the optional param, not hardcode it

(also I'm not convinced it's worth having more than one constructor)